invenio_rdm_records/services/communities/service.py

```diff
@@ -24,6 +25,39 @@ def create(self, identity, data, record, uow=None):
         pass
 
     @unit_of_work()
-    def delete(self, identity, id_, revision_id=None, uow=None):
+    def delete(self, identity, record_id, data, revision_id=None, uow=None):
         """Remove communities from the record."""
-        pass
+
+        record = self.record_cls.pid.resolve(record_id)
+        self.require_permission(identity, "remove_community_from_record", record=record)
+
+        valid_data, errors = self.schema.load(
+            data,
+            context={"identity": identity},
+            raise_errors=True,
+        )
+
+        removed_communities = []
+        errored_communities = []
+        community_ids = valid_data["communities"]
+        for community_id in community_ids:
+            if community_id not in record.parent.communities.ids:
+                errored_communities.append(
+                    {
+                        "community_id": community_id,
+                        "error": f"The record {record_id} does not belong to the community {community_id}",
+                    }
+                )
+                continue
+            record.parent.communities.remove(community_id)
+            uow.register(RecordCommitOp(record.parent))
+            removed_communities.append(community_id)
+
+        if not removed_communities:
+            raise RecordCommunityMissing(record_id, community_ids)
+
+        return_obj = {"removed communities": removed_communities}
+
+        if errored_communities:
+            return_obj["missing communities"] = errored_communities
```

Comment:
@ntarocco We need to alilgn on the names, these are probably not the best

Reply:
Personally, I like the approach of having a list of errors that can be validated by the client.

However, regarding the HTTP status code I don't have any strong opinion and I think it boils down on whether we see the list of modifications as atomic or not. For me, as soon as the HTTP resource (record) is modified, the HTTP code should be a `200`.

So if the modifications are not seen as atomic, I agree with @ntarocco suggestions. 

IMO, as an API consumer I typically like to see consistency, e.g. if bulk actions on other resources are atomic then I'd expect it to be the same here.

Reply:
I agree with the proposal for the responses, with some minor semantics for consistency:

- Always return 200 (even in complete failure). This is also consistent with our draft `PUT` behaviour. 
  - I assume this makes things easier/atomic in the UI?
- Errors as a list but with the following object structure:

```
{
  "errors": [
    { "community (or field)": "<community1_id>", "message": "error msg 1" },
    { "community (or field)": "<community2_id>", "message": "error msg 2" },
  ]
}, 200
```

Reply:
Thank you all for the feedback!
FYI: I implemented the proposal for the responses from @ntarocco with @slint observations